great_expectations/datasource/fluent/config_str.py

```diff
@@ -65,3 +67,26 @@ def __get_validators__(cls):
         # the value returned from the previous validator
         yield cls._validate_template_str_format
         yield cls.validate
+
+
+def _check_config_substitutions_needed(
+    datasource: Datasource,
+    options: Mapping,
+    raise_warning_if_provider_not_present: bool,
+) -> set[str]:
+    """
+    Given a Datasource and a dict-like mapping type return the keys whose value is a `ConfigStr` type.
+    Optionally raise a warning if config substitution is needed but impossible due to a missing `_config_provider`.
+    """
+    need_config_subs: set[str] = {
+        k for (k, v) in options.items() if isinstance(v, ConfigStr)
+    }
+    if (
+        need_config_subs
+        and raise_warning_if_provider_not_present
+        and not datasource._config_provider
+    ):
+        warnings.warn(
+            f"config variables '{','.join(need_config_subs)}' need substitution but no `_ConfigurationProvider` is present"
+        )
+    return need_config_subs
```

Comment:
@Kilo59 In one usage of `_check_config_substitutions_needed()` I saw, its return value is not used.  Hope it is OK.

Reply:
It's not being used here correct. I have an immediate follow-up PR where I am using it.